File: OOP C++/lab2/Dot.cpp
#include "Dot.h"
#include <stdexcept>
#include "FactoryInitializer.h"

void Dot::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand = operands.GetAndPop();
    output += std::to_string(operand);
    output += " ";
}

namespace {
    FactoryInitializer<Dot> Registration(".");
}

File: Lab1/src/Commands/Dot.java
package Commands;

import Utility.Operands;

import java.util.StringTokenizer;

public class Dot implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num = operands.GetAndPop();
        System.out.print(num + " ");
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}
